xchange-upbit/src/main/java/org/knowm/xchange/upbit/UpbitAdapters.java

```diff
@@ -130,28 +136,32 @@ public static Order adaptOrderInfo(UpbitOrderResponse upbitOrderResponse) {
       status = Order.OrderStatus.PARTIALLY_FILLED;
     } else if (upbitOrderResponse.getState().equalsIgnoreCase("done")) {
       status = Order.OrderStatus.FILLED;
+    } else {
+      status = Order.OrderStatus.NEW;
     }
-    OrderType type = upbitOrderResponse.getSide().equals("ask") ? OrderType.ASK : OrderType.BID;
-    BigDecimal originalAmount = upbitOrderResponse.getVolume();
-    String[] markets = upbitOrderResponse.getMarket().split("-");
-    CurrencyPair currencyPair =
-        new CurrencyPair(new Currency(markets[1]), new Currency(markets[0]));
-    String orderId = upbitOrderResponse.getUuid();
-    BigDecimal cumulativeAmount = upbitOrderResponse.getExecutedVolume();
-    BigDecimal price = upbitOrderResponse.getAvgPrice();
-    ZonedDateTime dateTime = ZonedDateTime.parse(upbitOrderResponse.getCreatedAt());
-    LimitOrder order =
-        new LimitOrder(
-            type,
-            originalAmount,
-            currencyPair,
-            orderId,
-            Date.from(dateTime.toInstant()),
-            price,
-            price,
-            cumulativeAmount,
-            upbitOrderResponse.getPaidFee(),
-            status);
-    return order;
+
+    final OrderType orderType = UpbitUtils.fromSide(upbitOrderResponse.getSide());
+    final CurrencyPair currencyPair = UpbitUtils.toCurrencyPair(upbitOrderResponse.getMarket());
+    return new LimitOrder.Builder(orderType, currencyPair)
+        .originalAmount(upbitOrderResponse.getVolume())
+        .id(upbitOrderResponse.getUuid())
+        .timestamp(Date.from(ZonedDateTime.parse(upbitOrderResponse.getCreatedAt()).toInstant()))
+        .limitPrice(upbitOrderResponse.getAvgPrice())
+        .averagePrice(upbitOrderResponse.getAvgPrice())
+        .cumulativeAmount(upbitOrderResponse.getExecutedVolume())
+        .fee(upbitOrderResponse.getPaidFee())
+        .orderStatus(status)
+        .build();
+  }
+
+  public static ExchangeMetaData adaptMetadata(List<UpbitMarket> markets) {
+    Map<CurrencyPair, CurrencyPairMetaData> pairMeta =
+        markets.stream()
+            .map(UpbitMarket::getMarket)
+            .map(UpbitUtils::toCurrencyPair)
+            .collect(
+                Collectors.toMap(
+                    Function.identity(), cp -> new CurrencyPairMetaData.Builder().build()));
```

Comment:
is there a reason why you aren't using cp here?

Reply:
The second argument of tomap is a function parameter, so I used cp only to receive an argument. 
Is there any way to recommend it?

Reply:
I was wondering more like why did you not pass in anything to the CurrencyPairMetaData Builders like .set(currencyPair) before building it, otherwise it doesn't have any data. Is that intentional? If yet that is fine leave it as is.

Reply:
Yes, it was intended.
CurrencyPairMetaData has no CurrencyPair field and no more values to fill in.